ibc-core/ics24-host/types/src/error.rs

```diff
@@ -9,6 +9,35 @@ use ibc_primitives::prelude::*;
 use ibc_primitives::proto::Error as ProtoError;
 use prost::DecodeError as ProstError;
 
+/// Errors that originate from host implementations.
+#[derive(Debug, Display)]
+pub enum HostError {
+    /// application module error: `{description}`
+    AppModule { description: String },
+    /// invalid data: `{description}`
+    InvalidData { description: String },
+    /// missing data: `{description}`
+    MissingData { description: String },
+    /// failed to update store: `{description}`
+    FailedToUpdateStore { description: String },
+    /// failed to store data: `{description}`
+    FailedToStoreData { description: String },
+    /// failed to retrieve data from store: `{description}`
+    FailedToRetrieveFromStore { description: String },
+    /// failed to parse data: `{description}`
+    FailedToParseData { description: String },
+    /// failed to validate client: `{description}`
+    FailedToValidateClient { description: String },
+    /// non-existent type: `{description}`
+    NonexistentType { description: String },
```

Comment:
Could be merged with the `MissingData` ?

Reply:
[817fa33](https://github.com/cosmos/ibc-rs/pull/1331/commits/817fa33cfab6b7e165d531565c1995b0d096d42c)